### ISSUE TITLE
Inference and Fine-tune

### ISSUE THREAD

**Issue by luoxue-star** [naver-ai/rope-vit#10]
Dear author, thank you for your excellent work. I have some questions regarding inference and fine-tune.

1. If the resolution of my image is (256, 256), do I need to fine-tune the model with an input size of 256? If not, how can I perform inference with an image size of (256, 256)?

2. If I need to fine-tune the model with an input size of 256, how should I set the initial learning rate (with a batch size of 32, one node, and four GPUs per node) and the number of fine-tuning epochs?

Thanks and best wishes!

**Comment by bhheo** [comment]
Hi

Thank you for your interest in our research

In the paper, we reported the performance without fine-tuning to a specific resolution

I have updated code
Now, you can do `256x256` inference with simple commands

```python
from models import vit_rope
model = vit_rope.rope_mixed_deit_small_patch16_LS(pretrained=True, img_size=256)
from models import swin_transformer_rope
model = swin_transformer_rope.swin_rope_mixed_tiny_patch4_window7_224(pretrained=True, img_size=256)
```

`img_size` should be specified for timm `PatchEmbed` layer, which has resolution check code

Note that you can also use `deit/` and `SwinTransformer/` code for ImageNet inference with `256 x 256`

- `deit/`
  ```bash
  OMP_NUM_THREADS=1 python -m torch.distributed.launch --nproc_per_node=8 --nnodes=1 --use_env main.py --model ${model_name} --finetune huggingface --data-path ${data_path} --output_dir ${save_path} --batch-size 128 --input-size 256 --eval --eval-crop-ratio 1.0 --dist-eval
  ```
- `SwinTransformer/`
  ```bash
  OMP_NUM_THREADS=1 python -m torch.distributed.launch --nproc_per_node=1 --nnodes=1 main.py \
  --eval --cfg configs/swinrope/${config_file} --resume huggingface --data-path ${data_path} --output ${save_path} --batch-size 128 --opts DATA.IMG_SIZE 256 MODEL.SWIN.WINDOW_SIZE 8
  ```

Please let me know if you have any further question

Best

**Comment by luoxue-star** [comment]
Thank you for your response! I do not have any questions at the moment. However, there is a bug in vit_rope.py when the image size is not equal to 224 and use absolute positional embedding, as the position embedding should be interpolated.

**Comment by bhheo** [comment]
Oh I missed ape models

I will fix it tomorrow
Thank you for report the bug

**Comment by bhheo** [comment]
The `ape_` model loading bug has been solved by https://github.com/naver-ai/rope-vit/commit/2e801b6a5cdb4a7a8d95ed555aef5645b58044df

Thank you for reporting the bug
